common/djangoapps/student/views.py

```diff
@@ -1718,7 +1722,7 @@ def create_account_with_params(request, params):
     # Perform operations within a transaction that are critical to account creation
     with transaction.atomic():
         # first, create the account
-        (user, profile, registration) = _do_create_account(form, custom_form)
+        (user, profile, registration) = _do_create_account(form, custom_form, site=getattr(request, 'site', None))
```

Comment:
What headaches do you see cropping up? This seems more robust than calling `request.site` since the down stack code does not require a site to be passed.